tests/unit/testServiceConfiguration.js

```diff
@@ -0,0 +1,397 @@
+    var oldModuleName = process.env.GAE_MODULE_NAME || '';
+    var oldModuleVersion = process.env.GAE_MODULE_VERSION || '';
+    var oldFunctionName = process.env.FUNCTION_NAME || '';
+
+    delete process.env.GAE_MODULE_NAME;
+    process.env.GAE_MODULE_VERSION = '1.0';
+    process.env.FUNCTION_NAME = 'someFunction';
+
+    var c = new Configuration({
+      serviceContext: {
+        service: 'customService'
+      }
+    }, logger);
+    t.deepEqual(c.getServiceContext().service, 'customService');
+    // The user didn't specify a version and thus because FUNCTION_NAME is 
+    // defined the version should be empty.
+    t.deepEqual(c.getServiceContext().version, '');
+
+    process.env.FUNCTION_NAME = oldFunctionName;
+    process.env.GAE_MODULE_VERSION = oldModuleVersion;
+    process.env.GAE_MODULE_NAME = oldModuleName;
+
+    t.end();
+  }
+);
+
+test(
+  'A Configuration uses the service name and version in the given config if ' +
+    'they were both specified and only the FUNCTION_NAME env var is set',
+  function(t) {
+    var oldModuleName = process.env.GAE_MODULE_NAME || '';
+    var oldModuleVersion = process.env.GAE_MODULE_VERSION || '';
+    var oldFunctionName = process.env.FUNCTION_NAME || '';
+
+    delete process.env.GAE_MODULE_NAME;
+    process.env.GAE_MODULE_VERSION = '1.0';
+    process.env.FUNCTION_NAME = 'someFunction';
+
+    var c = new Configuration({
+      serviceContext: {
+        service: 'customService',
+        version: '2.0'
+      }
+    }, logger);
+    t.deepEqual(c.getServiceContext().service, 'customService');
+    // The user specified version should be used
+    t.deepEqual(c.getServiceContext().version, '2.0');
+
+    process.env.FUNCTION_NAME = oldFunctionName;
+    process.env.GAE_MODULE_VERSION = oldModuleVersion;
+    process.env.GAE_MODULE_NAME = oldModuleName;
+
+    t.end();
+  }
+);
+
+test(
+  'A Configuration uses the service name "node" and a blank version if ' +
+  'GAE_MODULE_NAME is not set, FUNCTION_NAME is not set, and the user has '+
+  'not specified a service name or version',
+  function(t) {
+    var oldModuleName = process.env.GAE_MODULE_NAME || '';
+    var oldModuleVersion = process.env.GAE_MODULE_VERSION || '';
+    var oldFunctionName = process.env.FUNCTION_NAME || '';
+
+    delete process.env.GAE_MODULE_NAME;
+    delete process.env.GAE_MODULE_VERSION;
+    delete process.env.FUNCTION_NAME;
+
+    var c = new Configuration({}, logger);
+    t.deepEqual(c.getServiceContext().service, 'node');
+    t.deepEqual(c.getServiceContext().version, '');
+
+    process.env.FUNCTION_NAME = oldFunctionName;
+    process.env.GAE_MODULE_VERSION = oldModuleVersion;
+    process.env.GAE_MODULE_NAME = oldModuleName;
+
+    t.end();
+  }
+);
+
+test(
+  'A Configuration uses the service name "node" and a blank version if ' +
+  'GAE_MODULE_NAME is not set, FUNCTION_NAME is not set, and the user has '+
+  'not specified a service name or version even if GAE_MODULE_VERSION has ' +
```

Comment:
This case should not happen in theory. 
I am OK with the describe behavior. I would also be OK if the `GAE_MODULE_VERSION` was recorded as serviceContect.version.